lib/Dialect/ArithExt/IR/ArithExtOps.td

```diff
@@ -42,6 +43,13 @@ def ArithExt_MulOp : ArithExt_BinaryOp<"mul", [Commutative]> {
   }];
 }
 
+def ArithExt_MacOp : ArithExt_Op<"mac", [SameOperandsAndResultType, Pure, ElementwiseMappable]> {
+  let summary = "modular multiplication-and-accumulation operation";
+  let arguments = (ins SignlessIntegerLike:$lhs, SignlessIntegerLike:$rhs, SignlessIntegerLike:$acc, APIntAttr:$modulus);
+  let results = (outs SignlessIntegerLike:$output);
+  let hasVerifier = 1;
+  let assemblyFormat = "$lhs `,` $rhs `,` $acc attr-dict `:` qualified(type($output))";
```

Comment:
```suggestion
  let assemblyFormat = "operands attr-dict `:` type($output)";
```

Reply:
You could use this for all the others as well. I have started defaulting to having the base class op define an assembly format as

```
operands attr-dict `:` functional-type(operands, results)
```

And then override the part after the `:` for type-inferrible ops. Generally this gives a good format without having to specify anything op-specific.